capa/features/extractors/dnfile/helpers.py

```diff
@@ -107,8 +107,18 @@ def format_name(module, method):
         return f"{module}.{method}"
 
 
+def validate_has_dotnet(pe: dnfile.dnPE):
+    assert pe.net is not None
+    assert pe.net.mdtables is not None
+    assert pe.net.Flags is not None
+
+
 def resolve_dotnet_token(pe: dnfile.dnPE, token: Token) -> Any:
     """map generic token to string or table row"""
+    validate_has_dotnet(pe)
+    assert pe.net is not None
+    assert pe.net.mdtables is not None
+
     if isinstance(token, StringToken):
```

Comment:
@mike-hunhoff this is rather verbose and annoying: we have to convince mypy that potentially optional .NET fields are not None and we have to do it everywhere.

the best alternative that i've come up with is to introduce a new type (and mess with the dnfile.dnPE constructors) that is guaranteed to have `pe.net` and `pe.net.mdtables` once some validation is done. im not sure if its worth the gymnastics there or just have extract assertions in some of these routines.

i've used `assert` because it will fire at runtime if we try to access something that doesn't exist. really, we'd want a more specific exception, but since it indicates a programming error (we thought the field access was valid), i think AssertionError is ok and will lead to sufficient bug reports.

thoughts?